DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Geometry/ISpeckleObjectToHost/ArcToHostConverter.cs

```diff
@@ -4,51 +4,46 @@
 
 namespace Speckle.Converters.ArcGIS3.Geometry.ISpeckleObjectToHost;
 
-//TODO: Ellipses don't convert correctly, see Autocad test stream
-//[NameAndRankValue(nameof(SOG.Arc), NameAndRankValueAttribute.SPECKLE_DEFAULT_RANK)]
-public class CurveToHostConverter : IToHostTopLevelConverter, ITypedConverter<SOG.Arc, ACG.Polyline>
+[NameAndRankValue(nameof(SOG.Arc), NameAndRankValueAttribute.SPECKLE_DEFAULT_RANK)]
+public class ArcToHostConverter : IToHostTopLevelConverter, ITypedConverter<SOG.Arc, ACG.Polyline>
 {
   private readonly ITypedConverter<SOG.Point, ACG.MapPoint> _pointConverter;
+  private readonly IConversionContextStack<ArcGISDocument, ACG.Unit> _contextStack;
 
-  public CurveToHostConverter(ITypedConverter<SOG.Point, ACG.MapPoint> pointConverter)
+  public ArcToHostConverter(
+    ITypedConverter<SOG.Point, ACG.MapPoint> pointConverter,
+    IConversionContextStack<ArcGISDocument, ACG.Unit> contextStack
+  )
   {
     _pointConverter = pointConverter;
+    _contextStack = contextStack;
   }
 
   public object Convert(Base target) => Convert((SOG.Arc)target);
 
   public ACG.Polyline Convert(SOG.Arc target)
   {
-    // Determine the number of vertices to create along the arc
-    int numVertices = Math.Max((int)target.length, 50); // Determine based on desired segment length or other criteria
-    List<SOG.Point> pointsOriginal = new();
-
-    // get correct direction
-    double? angleStart = target.startAngle;
-    double? fullAngle = target.endAngle - target.startAngle;
-    double? radius = target.radius;
-
-    if (angleStart == null || fullAngle == null || radius == null)
+    if (
+      target.plane.normal.x != 0 || target.plane.normal.y != 0 || target.plane.xdir.z != 0 || target.plane.ydir.z != 0
```

Comment:
Here it looks like you're checking for xy planarity on the world xy plane: if arcgis can't support xy planar curves on other planes (z is non-zero), then the exception message should be more specific since `2d arc` can mean planar arcs on planes in any 3d orientation

Reply:
Also, just a caution that we can't guarantee that the plane property is always accurate: the absolute safest way to test for planarity is to check the start and end points